File: src/store/reducer/userReducer.js
import {GET_USER_INFO, LOGIN_SUCCESS, LOGOUT_SUCCESS} from '../types';

const initialState = {
  isLogin: false,
  userInfo: {},
};

export default (state = initialState, action) => {
  switch (action.type) {
    case GET_USER_INFO:
      return {
        ...state,
        userInfo: action.payload,
      };
    case LOGIN_SUCCESS:
      return {
        ...initialState,
        isLogin: action.payload,
      };
    case LOGOUT_SUCCESS:
      return {
        ...state,
        isLogin: action.payload,
      };
    default:
      return state;
  }
};

File: lib/store/reducer/userReducer.dart
import '../type.dart';
import '../modal/User.dart';

User userReducer(state, action) {
  switch (action.type) {
    case Type.LOGIN_SUCCESS:
      return User(name: action.payload.name, isLogin: true);
      break;
    case Type.LOGIN_OUT_SUCCESS:
      return User.init();
      break;
    default:
      return state;
  }
}
